Dear Editor,

Please find attached our manuscript, *Gene duplications potentiate adaptive evolution of complex traits*, for your consideration.

**What is the scientific question you are addressing?**

How does gene duplication influence the evolution of biological complexity?
Specifically, we examine the role of adative processes (e.g., neo-functionalization, dose effects) in shaping gene duplication's impact on the measures of biological complexity.

**What is the key finding that answers this question?**

We find that gene duplication promotes *de novo* evolution of complex phentoypic traits via neo-functionalization.
This effect is limited to complex traits, with adaptive evolution of simpler traits exhibiting no benefit beyond the effect of increased genome length alone.
<!-- We find the adaptive benefit of gene duplication to be strongest in enabling adaptive evolution of complex, rather than simple, traits. -->
Through step-by-step analysis of lineage histories, we trace an outsized fraction of coding sites for novel complex traits back to duplicated regions; similarly, this effect is also specific to complex traits, not being observed for simple traits.

**What is the nature of the evidence you provide in support of your conclusion?**

We report *in silico* evolution experiments comparing the adaptive evolution of complex traits both with and without of duplicative mutation processes, in conjunction with lineage-based analyses examining the genomic origins of evolved traits.
For this purpose, we used the Avida digital evolution platform, which enables self-replicating computer programs to be used as a model organism for evolution experiments.
To study phenotypic complexity, a framework for adaptive traits within Avida exeriments drawn from Lenski et al. (2003).
This framework defines a set of available metabolic functions with varying complexity based on the number of logical operations necessary to carry them out.

While careful consideration must be taken in generalizing results from Avida to biological organisms, in the context of our work, this model offers unique capabilities for detailed, rigorously controlled treatments with populations undergoing full Darwinian evolution.
Our computational medium enabled further work testing partial ablations of gene duplication mechanisms to separate the impact of propagating genetic information from side effects in increased genome length and increased effective mutation rate.
We also take advantage of detailed lineage records to examine exact mutational histories of coding sites for phenotypic traits to discern the role of gene duplication.

> Lenski, R. E., Ofria, C., Pennock, R. T., & Adami, C. (2003). The evolutionary origin of complex features. Nature, 423(6936), 139–144. https://doi.org/10.1038/nature01568

**What are the three most recently published articles that are relevant to this question?**

> Mihajlovic, L., Iyengar, B. R., Baier, F., Barbier, I., Iwaszkiewicz, J., Zoete, V., Wagner, A., & Schaerli, Y. (2025). A direct experimental test of Ohno’s hypothesis. eLife, 13, RP97216. doi:10.7554/eLife.97216
elifesciences.org

Compares rate of improvement for coGFP floursescence strength between single- vs. dual-copy populations under directed evolution conditions in an *E. coli* model.
The two-copy strain does not exhibit adaptive advantage, contrasting notably with present work.

> Tong, K., Datta, S., Cheng, V., Haas, D. J., Gourisetti, S., Yopp, H. L., Day, T. C., Lac, D. T., Conlin, P. L., Bozdag, G. O., & Ratcliff, W. C. (2025). Genome duplication in a long-term multicellularity evolution experiment. Nature, 639, 691–699. doi:10.1038/s41586-025-08689-6

Demonstrates adaptive role of dose effects from whole genome duplication, using experimental evolution in *S. cerevisiae* model.
By contrast, our focus pertains to neofunctionalization and evolutionary potentiation.

> DuBose, J. G., & de Roode, J. C. (2024). The link between gene duplication and divergent patterns of gene expression across a complex life cycle. Evolution Letters, 8(5), 726–734. doi:10.1093/evlett/qrae028

Applying a comparative approach establishes evidence for adaptive neo-functionalization of duplicated genes in monarch butterflies vis-a-vis shifts in expression patterns between lifecycle stages.
Our experiments generally echo these findings, with newly evolved complex tasks tending to arise within duplicated segments.

**What significance do your results have for the field?**

Our study delivers direct experimental evidence of gene duplication increasing evolved complexity via adaptive innovation in complex traits.
By dissecting duplicative mutational mechanisms in Avida and tracing detailed mutational steps along complete lineages, we demonstrate that:
1. gene duplications can facilitate *de novo* origin of adaptive traits,
2. this effect is pronounced for complex traits, and
3. genetic information content propagated in duplications contributes to this advantage.

These findings contrast notably with hypotheses that discount adaptive explanations for biological complexity.
<!-- As such, this work provides a cornerstone complementary to comparative approaches in natural history. -->

**What significance do your results have for the broader community (of biologists and/or the public)?**

We contribute to the body of evidence highlighting the role of duplicative processes in adaptive evolution.
Our findings pertain to application-oriented work harnessing evolution as an optimization process, both *in vivo* and *in silico* (e.g., directed evolution or evolutionary computation).
In particular, our research identifies problem contexts where gene duplication processes may be most impactful (e.g., complex problems with potential to assemble functional building blocks).

From a science communication perspective, lines of research working with ``evolution in action'' provide exciting opportunity to broaden public perceptions of evolutionary processes as an ongoing story rather than a historical record.
In concert with the broader experimental evolution community, digital evolution work is particularly well-suited to ``hands-on'' exploration --- especially through web-based applications, such as Avida-ED (<https://avida-ed.msu.edu>).

**What other novel findings do you present?**

- Propagation of intact genetic sequence content by slip duplication yielded markedly larger final genome size, compared to ablation treatments where duplications did not preserve sequence content --- including treatments where inserts comprised fully neutral content.
- Ablation experiments also demonstrated that propagation of intact genetic sequence content contributes in facilitating adaptive innovation.
- Slip duplication accelerated the accumulation of vestigial coding sites in the genome, beyond the impact of increased genome size alone.

**Is there additional information that we should take into account?**

Our project builds on an impactful body of experimental work using Avida to study the evolution of complexity, notably including two works appearing in PLOS Biology --- Zaman et al. 2014 (doi:[10.1371/journal.pbio.1002023](https://doi.org/10.1371/journal.pbio.1002023)) and Goldsby et al. 2014 (doi:[10.1371/journal.pbio.1001858](https://doi.org/10.1371/journal.pbio.1001858)).
Note also that an earlier version of this work appeared at the 2017 Artificial Life conference (doi:[10.1162/isal_a_045](https://doi.org/10.1162/isal_a_045)).
By presenting our findings in PLOS Biology, we look forward to furthering conversation around the evolution of complexity.

Thank you for your consideration,
The Authors
