\section{Conclusion} \label{sec:conclusion}

How information is represented within genomes and the evolutionary processes through which it accumulates is a fundamental pillar of biological science \citep{adami2024evolution}.
Among other topics, important questions remain about the evolutionary role in structuring and extending genetic information.
One challenge to developing a rigorous and detailed understanding of gene duplications is the difficulty of realizing biological experiments to explore a counterfactual case absent gene duplication.
By leveraging an \textit{in silico} study system that may be arbitrarily manipulated and exactly observed, our experimental approach seeks to deepen and systematize our understanding of the evolutionary consequences of gene duplications.

% Recap of what we found:
Across surveyed partial analogs of gene duplication, we find that both sequence content and order contribute to facilitating adaptive evolution.
While simply increasing supply of raw genetic material facilitates the evolution of simple traits, we found duplicating existing genome information to uniquely facilitate evolution of complex traits.
In our experiments, duplications significantly potentiate genome regions that go on to code for novel complex traits but --- consistent with neofunctionalization theory --- do not potentiate simple traits.
From a genetic architecture perspective, we do not find direct evidence of increasing genome brittleness via subfunctionalization processes under slip duplication; however, we found slip duplication to accelerate the accumulation of coding material in genomes when vestigial sites are considered.

One intrinsic limitation of working with a digital study system is in discerning which aspects of behavior within our experiments generalize to natural systems.
Nonetheless, such work can help establish the sufficiency of theory to explain empirical phenomena.
Such work can also inform questions around universal properties of life \citep{dorin2024what}, pertinent to questions of artificial life or astrobiology.

% future work:
Results presented here motivate several future studies.
Performing deeper, play-by-play analyses of individual lineages evolved with gene duplications could allow us to identify examples of subfunctionalization or neofunctionalization, and deepen our understanding of  their contributions to promoting the evolution of complex features through a case study approach \citep{mcphee2018detailed}.
Future work might also extend our retrospective approach for detecting potentiation into full-fledged replay experiments, where replicate evolutionary trajectories are collected to directly identify the statistical impact of evolutionary events in influencing later evolutionary outcomes \citep{blount2018contingency,Ferguson2023}.
Such efforts will further help establish the role that gene duplication plays in natural evolution and could play in computational systems.

% application-oriented outcomes:
While evolutionary studies are essential for understanding the natural world around us, evolution can also be harnessed for applied purposes.
Within the domain of evolutionary computation.
Findings from our work can help guide the development of more effective algorithms for evolutionary computation, where evolution-inspired approaches are applied to real-world engineering challenges \citep{holland1992genetic}.
Aspects of these findings may also generalize to bioengineering via directed evolution, in informing approaches to targeted mutagenesis \citep{sandberg2019emergence}.

% Other ideas:
% * What effects are deletions having in out slip mutation operators?
% * Recognize that there are many more variants we could try to tease more aspects of gene duplication apart: more slip-scatter variants (to better test if it matters that inserted segments are contiguous), does it matter that inserted segments are adjacent to the target segment? What if we run a continuum of operators between slip-duplicate and slip-scramble? Slip-duplicate with chance of other types of mutations?
