\section{Results and Discussion} \label{sec:results}

% @RCK - writing group
% Structure your R&D section to say, for each hypothesis 1. What you're showing. 2. Here's the evidence.
% Combine these figures to support your conclusions.
%
% Think about making two-part figures, one with the time-series, and then a partner panel that has the final dominant comparisons (violin or box plots).
%
% Visually distinguish your figures so that the reader, at a glance, can tell what they're looking at. Consider color-coding the backgrounds, or having distinct plot types for different kinds of conclusions.
%
% If you MUST show a pile of time series, consider combining them into sets of panels. Say one big figure with all your timeseries, but quadrants showing each one as part A, B, C, etc.
%
% Typesetting point: italicize or bold or underline key points or terms, particularly treatment names

\subsection{Does gene duplication promote evolvability?}
We found that gene duplication promotes evolvability in all three environments. As shown in Figure \ref{fig:results_panels}, organisms evolved in the slip-duplicate treatment had significantly higher phenotypic match scores than organisms evolved in the baseline treatment across the static, simple changing and complex changing environments (two-tailed Mann-Whitney U tests, W = 562.5, 2702.5, 1540 respectively, Bonferroni-adjusted $p$ values all $<< 0.0001$). This result suggests that the slip-duplicate operator promotes the evolution of complexity
%two qualitatively different types of environments: 1) in a
both in static environments that require the evolution %and maintenance
of unconditional task performance (Boolean logic operations), and in dynamic environmental conditions that require the evolution of regulatory mechanisms that alter expression based on current environmental conditions.
These results, as expected, support the existing literature on the capacity of gene duplications to promote evolvability %for both unregulated and regulated traits
\citep{Koza:1995fr,Zhang:2003fw,Teichmann:2004cz}. % MJW: This seems like an odd sentence without citations. -- AML: Indeed, it does...

% AML: Still not really sure how we want to present the results panels. Thoughts?
% MJW: The lower figure in each panel (which is an odd formatting choice: this is really a 6 panel figure you're treating as 3) does not need to be anywhere near the width it is.  The point of those time graphs, as far as I can tell, is to show that the end point is not inherently weird compared to the rest of the trajectory.  How about trying putting the two graphs of each panel side by side, possibly even by having the violin plot panels take up 2/3 of the horizontal real estate and/or maybe replacing the categorical x axis labels with 1, 2, 3 etc and explaining them in the caption?

\input{fig/results_panels.tex}

\subsection{What configuration of gene duplication is most evolvable?}
Across all three environments
%(static, simple changing, and complex changing),
we found that the most important aspect of gene duplications is the capacity to duplicate meaningful information in the genome. As shown in Figure \ref{fig:results_panels}, %across all three environments,
only organisms that evolved in the slip-duplicate and slip-scramble treatments had significantly higher phenotypic match scores than organisms evolved in the baseline treatment.
Organisms evolved in all other experimental treatments were either not significantly different from the baseline treatment or significantly worse than the baseline treatment.

Of the five types of slip mutation operators used (see Figure \ref{fig:slip_mut_variants}), only the slip-duplicate and slip-scramble operators were capable of duplicating meaningful information in a genome.
Slip-scramble mutations maintain the content but not the ordering of duplicated instruction sequences; thus, they can duplicate information about what instructions make-up already successful genetic sequences but do not maintain the particular ordering of those instructions. Slip-duplicate mutations are the full form of gene duplication, %capable of duplicating even more information than slip-scramble mutations,
able to exactly duplicate the ordered instruction make-up of existing genetic sequences. In contrast to the slip-scramble and slip-duplicate operators, all other slip mutation operators do not duplicate information about instruction sequences already present in the genome.

Because the slip-scramble treatment was significantly better at promoting evolvability than the baseline treatment, we can conclude that the duplication of functional building blocks is valuable no matter how they are arranged when duplicated. However, is %duplicating more information better? Does it matter that gene duplications
it important to
preserve both the contents and ordering of duplicated sequences?

\subsubsection{Is more information better?}
From Figure \ref{fig:results_panels}, organisms evolved in the slip-duplicate treatment had higher phenotypic match scores than organisms evolved in the slip-scramble treatment, implying that the additional information in a gene duplication event %is able to replicate in a genome, the more it will
does
promote evolvability. To confirm this relationship and reduce multiple comparison issues with statistical tests, we re-ran 100 new trials of both the slip-duplicate and slip-scramble treatments %with new random number seeds
and compared the phenotypic match scores of the evolved organisms. In all three environments, organisms that evolved in the slip-duplicate treatment had significantly higher phenotypic match scores than organisms evolved in the slip-scramble treatment (two-tailed Mann-Whitney U tests, W = 4305, 4028.5, 3621.5 respectively, Bonferroni-adjusted $p$ values 0.0109, 0.0222, 0.0017, respectively). This result supports that %the capacity to duplicate more information is better, and the fact that gene duplications are capable of
duplicating both the content and structure of existing genetic code is an important factor in their ability to promote evolvability.

\subsection{Discerning the mechanism of slip duplication in facilitating adaptive evolution}

Having established slip-duplicate as the best-performing slip insertion operator, we next sought to understand the mechanism driving the evolvability impact of gene duplication in the Avida system.
In follow-on investigations, we next (1) identify the characteristics of phenotypic traits that are most facilitated by slip duplication, (2) identify signatures of in slip duplication in potentiating these traits in lineage histories, and (3) investigate how slip duplication affects the net accumulation of brittleness and vestigial genetic material in genomes.

\subsubsection{Slip duplication facilitates adaptive variation for the evolution of complex traits.}

One salient aspect of the task system in Avida is the hierarchical composition of task definitions.
``Simpler'' tasks like AND and NAND form the building blocks of the definition of ``more complex'' tasks like XOR and EQU.
In the past, this property has been applied to study the evolution of complex traits \citep{TODO}.
The complexity of the nine tasks used in this work can be approximted by the minimal number of NAND buliding blocks required to perform them\footnote{
In computer architecture theory, NAND is commonly established as the fundamental building block operation as all other logical operations are able to be derived from compositions of NAND gates \citep{TODO}.
},
\begin{enumerate}
\item NOT and NAND: one NAND component,
\item AND and ORNOT: two NAND components,
\item OR and ANDNOT: three NAND components,
\item NOR and XOR: four NAND components,
\item EQU: five NAND components.
\end{enumerate}
We exploited this natural hierarchy to test how gene duplication facilitated evolution across a spectrum of task complexity, by studying the rate at which different ``component count'' categories of tasks evolved.

One simple consequence of gene duplication is in increasing the number of sites in a genome.
We sought to identify the aspects of gene duplication evolvability that could be attributed simply to longer genome length.
For this purpose, we compared against an additional control, the long genome baseline control where genomes were 1,000 sites long.


\input{fig/adaptive-evolution-rate.tex}

In a first set of experiments, we tested whether the adaptive boost of slip duplication was greater than the adaptive boost of simply having more genome sites.
Figure \ref{fig:adaptive-evolution-rate} shows the results of these experiments.

Genome length was sufficient to explain the boost to the rates at which simpler traits ($\leq 3$ components) were gained in the slip duplication treatment compared to the baseline treatment --- the long-genome baseline treatment indeed faster adaptive evolution of these traits compared to the slip-duplication treatment.
% https://github.com/chaynes2019/AvidaGeneDupe/blob/538ede79c7301f10718ca96c8dd38782b6882632/binder/adaptive-evolution-rate.ipynb
However, for 4 and 5 component traits, the slip-duplicate treatment had a significantly higher fraction of replicates that had evolved the these traits at 600 generations compared to the long genome control (exact Fisher tests; 24/60 vs. 36/60 $p<0.05$ [4 components], 2/30 vs. 10/30 $p < 0.03$ [5 components]; Figure \ref{fig:adaptive-evolution-rate:direct}).

\subsubsection{Slip-duplicated genome regions are potentiated for novel complex traits}

% https://github.com/chaynes2019/AvidaGeneDupe/blob/4c7fa27229094adcb5bdb0b1aec541d0014b0fed/binder/hard-task-gain.ipynb
%             H-statistic       p-value
% Components
% 1              1.011820  3.144672e-01
% 2             22.787798  1.809107e-06
% 3             33.846753  5.962854e-09
% 4             15.359894  8.885441e-05
% 5              9.097744  2.559249e-03
%    Components  Prev Slip Insertion Cumulative Count      mean       std
% 0           1                                 False  0.932088  0.455109
% 1           1                                  True  1.002025  1.164197
% 2           2                                 False  0.588566  0.618479
% 3           2                                  True  1.579974  1.191681
% 4           3                                 False  0.491481  0.700256
% 5           3                                  True  1.516115  0.911430
% 6           4                                 False  0.703793  0.980105
% 7           4                                  True  1.229158  0.654042
% 8           5                                 False  0.583674  0.927917
% 9           5                                  True  1.249985  0.531423
%    Components  Prev Slip Insertion Cumulative Count  size
% 0           1                                 False    60
% 1           1                                  True    48
% 2           2                                 False    60
% 3           2                                  True    56
% 4           3                                 False    60
% 5           3                                  True    59
% 6           4                                 False    52
% 7           4                                  True    52
% 8           5                                 False    20
% 9           5                                  True    20



\input{fig/potentiation}

\subsubsection{Role of slip duplication in facilitating robustness and raw neutral genetic material.}

\input{fig/num-coding-sites}

We first investigated whether gene duplication increased genome robustness.
One definition of robustness is in terms of brittleness: the number of genome sites ``critical'' to fitness, where a single-site knockout caused the loss of one or more tasks.
We measured genome robstness through all-site knockouts using Avida's genome evaluator.


We found that instead of decreasing the amount of brittleness, genomes with slip duplications had comparable amounts of brittleness to the long-genome baseline treatment.
However, we found that the slip duplications provided a higher amount of previously coding material that was now neutral with respect to task performance.
So, in a sense a greater amount of coding material was available for neutral re-shaping even though overall robustness was not increased
% https://github.com/chaynes2019/AvidaGeneDupe/blob/538ede79c7301f10718ca96c8dd38782b6882632/binder/indel-effect-nulldist.ipynb
In additional experiments, we tested the effects of insertion mutations on brittleness (measured by coding site count) and found that neutral insertion mutations, indeed, do tend to reduce coding site count.
On average, insertion mutations that do not affect task count decrease coding site count by 6.8 sites.
This effect is strongest in genomes with high complexity; for instance, neutral insertion mutations decrease coding site count by 9.2 and 8.3 sites, on average, in genomes that encode 4 and 5-component complexity tasks, respectively.
Supplementary Figure \ref{fig:nulldist} compares distributions of phenotypic and coding site outcomes across lineages from across our experiments.
One possible explanation would be that selective pressures are driving the two treatments to reach the same ballpark of critical sites, but the slip duplication can duplicate and copy around genome content in a manner that increases the copy count of previously-coding sites relative to the number of critical sites.

(It should be noted that the coding site history measure is limited, as it depends on functional sites being detectable at some point through single-site knockout; such an approach can fail to reveal aspects of sequence complexity \citep{moreno2024cryptic}.)

\subsection{Major Results}

In this work, we have found the most effective form of gene duplication in facilitating adaptive evolution within the Avida platform was the direct, local slip mutation.

First, we sought to identify which aspects of adaptive evolution, in particular, are facilitated by slip duplication.
We found that, relative to a control experiment for the impact of slip duplication on increasing genome size, gene duplication most greatly benefited the evolution of complex traits.

Then, we turned to a mechanistic lens to assess whether the system aligns with expectations from neofunctionalization theory.
We found that, indeed, that the raw material created by slip duplication played a potentiating role in the evolution of complex traits.
Specifically, we identified that slip-duplicated regions are significantly more likely to host coding sites for new traits when they first arise.
In line with neofunctionalization expectations, however, we do not observe any potentiation effects of slip duplication on the evolution of very simple traits.

In a final exploration, we turn to the questions around genome architecture, to assess the consequences of slip duplication with respect to genome complexity.
One idea in this realm is that gene duplication might increase the rate at which genetic complexity grows by providing raw material for contingent complexity to arise as drift effects whittle back the redundancies introduced by slip duplication.
In contrast to this expectation, though, we find the rate of growth in genome complexity to be in line with that observed under control experiments with slip duplications disabled.

However, we do observe a significant increase in the rate at which material that previously coded for phenotypic traits accumulates in the genome under slip duplication.
In assessing the reason how this might occur, we tested the immediate-term effects of slip duplication on genome brittleness.
We find that in the immediate aftermath neutral slip duplications on average decrease, rather than increase, the number of critical coding sites in a genome.
These results concord and shed light on the increase in previously-coding material in genomes.
Ostensibly these effects in reducing brittleness are balanced entropic effects in resulting in a similar trajectory of overall genome complexity between slip duplication treatments and controls.

% outro
Much theoretical interest is paid to the role of gene duplication in facilitating neutral evolution, and opening the door for adaptive evolution of new complexity through subfunctionalization and neofunctionalization.
Indeed, ample evidence from experiments \textit{in silico} and \textit{in vivo} has been collected to support this view of the role of gene duplication in adpative evolution.

In contrast, our study highlights the with greater detail and specificity the consequences of gene duplication:
\begin{enumerate}
\item while increasing the amount of raw genetic material available to work with suffices in facilitating the evolution of simple traits, duplication of existing genome material in particular is more effective at facilitating the evolution of complex traits;
\item mechanistically, we find duplicated genome materials are significantly potentiated to code for novel complex traits, but there is no effect for simple traits;
\item from a genetic architecture perspective, we find that genome brittleness does not increase significantly under slip duplication compared to control treatments; however, slip duplication accentuates the rate at which raw material that coded for traits accumulates.
\end{enumerate}

% \subsection{High mutation rate inhibits the evolution of regulation in changing environments}
% In addition to our results on why gene duplications promote evolvability, our experiments
% %demonstrate the potential for
% indicate that
% high mutation rates inhibit the evolution of regulation in changing environments. In both the simple and complex changing environments, organisms evolved in the high mutation rate treatment had significantly lower phenotypic match scores than organisms evolved in the baseline treatment, which can be seen in Figure \ref{fig:results_panels} (two-tailed Mann-Whitney U tests, W = 9958 and 9944, respectively, Bonferroni-adjusted $p$ values both $<< 0.0001$).
% A similar result was found in \citep{Lalejini:2016plasticity}. Although higher mutation rates increase genetic variation, most mutations have deleterious effects \citep{Schlichting:2002jl}. Thus, higher mutation rates may increase the difficulty of maintaining the genetic machinery necessary for complex regulation.

% If elevated mutation rates %can work to
% inhibit the evolution of regulation and all experimental treatments with slip mutation operators have an effectively higher mutation rate than the baseline treatment, why do we see this effect only in the high mutation rate treatment? The rates of insertions and deletions per instruction copied in the high mutation rate treatment were selected to, on average, result in approximately the same number of mutations per divide as in our treatments with slip mutations. Yet, only organisms evolved in the high mutation rate treatment had significantly lower phenotypic match scores than organisms evolved in the baseline treatment in the changing environments; all other experimental treatments were either significantly better or not significantly different than the baseline treatment.

% @amlalejini old text on mutation load explanation:
%Why is it that \textit{only} the high mutation rate treatment inhibited evolvability in the changing environments? One possibility is because slip mutations result in large mutational events that affect few offspring, whereas mutations in the high mutation rate treatment result in an increased number of smaller mutational events that are spread across many offspring. Thus, many offspring in the high mutation rate treatment are subject to the increased rate of deleterious mutations that results from an increase in the rates of copy insertions and copy deletions. This is in contrast to treatments with slip mutations where relatively fewer offspring are subjected to the increased rate of deleterious mutations because deleterious mutations are concentrated in a fewer offspring relative to the high mutation rate treatment.
%Further analysis is needed to confirm that this is, indeed, the case.
% @amlalejini new text on mutation load explanation:
%  - should I cite something for 'mutation load' (i.e. this concept exists, read about it here)?
% Why is it that \textit{only} the high mutation rate treatment inhibited evolvability in the changing environments?
% One possibility stems from the higher mutation load imposed on populations by the high mutation rate treatment.
% Slip mutations result in large mutational events that affect few offspring, whereas the elevated rate of copy mutations in the high mutation rate treatment results in an increased number of smaller mutational events spread across many offspring.
% Thus, many offspring in the high mutation rate treatment are subject to the increased rate of deleterious mutations that results from the higher rates of copy insertions and copy deletions. This is in contrast to treatments with slip mutations where relatively fewer offspring are subjected to large mutational events, which results in the concentration of deleterious mutations in fewer offspring relative to the high mutation rate treatment.
% Further analysis is needed to confirm that this is, indeed, the case.

% OLD figures
% \input{fig/FullReplicationTaskCountTimeCourse.tex}
% \input{fig/FullReplicationFinalDominantTaskCountViolin.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantInfo.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantInfoConcentration.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantGenomeLengthsViolin.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantCodingSitesPerTask.tex}
% \input{fig/ModifiedReplicationFinalDominantTaskCountViolin.tex}
% \input{fig/LalejiniEtAlReplicationRealFinalDominantCodingSitesPerTask.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantOverlap.tex}
