\section{Results and Discussion} \label{sec:results}

% @RCK - writing group
% Structure your R&D section to say, for each hypothesis 1. What you're showing. 2. Here's the evidence.
% Combine these figures to support your conclusions.
%
% Think about making two-part figures, one with the time-series, and then a partner panel that has the final dominant comparisons (violin or box plots).
%
% Visually distinguish your figures so that the reader, at a glance, can tell what they're looking at. Consider color-coding the backgrounds, or having distinct plot types for different kinds of conclusions.
%
% If you MUST show a pile of time series, consider combining them into sets of panels. Say one big figure with all your timeseries, but quadrants showing each one as part A, B, C, etc.
%
% Typesetting point: italicize or bold or underline key points or terms, particularly treatment names

\subsection{Does gene duplication promote evolvability?}
We found that gene duplication promotes evolvability in all three environments. As shown in Figure \ref{fig:results_panels}, organisms evolved in the slip-duplicate treatment had significantly higher phenotypic match scores than organisms evolved in the baseline treatment across the static, simple changing and complex changing environments (two-tailed Mann-Whitney U tests, W = 562.5, 2702.5, 1540 respectively, Bonferroni-adjusted $p$ values all $<< 0.0001$). This result suggests that the slip-duplicate operator promotes the evolution of complexity
%two qualitatively different types of environments: 1) in a
both in static environments that require the evolution %and maintenance
of unconditional task performance (Boolean logic operations), and in dynamic environmental conditions that require the evolution of regulatory mechanisms that alter expression based on current environmental conditions.
These results, as expected, support the existing literature on the capacity of gene duplications to promote evolvability %for both unregulated and regulated traits
\citep{Koza:1995fr,Zhang:2003fw,Teichmann:2004cz}. % MJW: This seems like an odd sentence without citations. -- AML: Indeed, it does...

% AML: Still not really sure how we want to present the results panels. Thoughts?
% MJW: The lower figure in each panel (which is an odd formatting choice: this is really a 6 panel figure you're treating as 3) does not need to be anywhere near the width it is.  The point of those time graphs, as far as I can tell, is to show that the end point is not inherently weird compared to the rest of the trajectory.  How about trying putting the two graphs of each panel side by side, possibly even by having the violin plot panels take up 2/3 of the horizontal real estate and/or maybe replacing the categorical x axis labels with 1, 2, 3 etc and explaining them in the caption?

\input{fig/results_panels.tex}

\subsection{What configuration of gene duplication is most evolvable?}
Across all three environments
%(static, simple changing, and complex changing),
we found that the most important aspect of gene duplications is the capacity to duplicate meaningful information in the genome. As shown in Figure \ref{fig:results_panels}, %across all three environments,
only organisms that evolved in the slip-duplicate and slip-scramble treatments had significantly higher phenotypic match scores than organisms evolved in the baseline treatment.
Organisms evolved in all other experimental treatments were either not significantly different from the baseline treatment or significantly worse than the baseline treatment.

Of the five types of slip mutation operators used (see Figure \ref{fig:slip_mut_variants}), only the slip-duplicate and slip-scramble operators were capable of duplicating meaningful information in a genome.
Slip-scramble mutations maintain the content but not the ordering of duplicated instruction sequences; thus, they can duplicate information about what instructions make-up already successful genetic sequences but do not maintain the particular ordering of those instructions. Slip-duplicate mutations are the full form of gene duplication, %capable of duplicating even more information than slip-scramble mutations,
able to exactly duplicate the ordered instruction make-up of existing genetic sequences. In contrast to the slip-scramble and slip-duplicate operators, all other slip mutation operators do not duplicate information about instruction sequences already present in the genome.

Because the slip-scramble treatment was significantly better at promoting evolvability than the baseline treatment, we can conclude that the duplication of functional building blocks is valuable no matter how they are arranged when duplicated. However, is %duplicating more information better? Does it matter that gene duplications
it important to
preserve both the contents and ordering of duplicated sequences?

\subsubsection{Is more information better?}
From Figure \ref{fig:results_panels}, organisms evolved in the slip-duplicate treatment had higher phenotypic match scores than organisms evolved in the slip-scramble treatment, implying that the additional information in a gene duplication event %is able to replicate in a genome, the more it will
does
promote evolvability. To confirm this relationship and reduce multiple comparison issues with statistical tests, we re-ran 100 new trials of both the slip-duplicate and slip-scramble treatments %with new random number seeds
and compared the phenotypic match scores of the evolved organisms. In all three environments, organisms that evolved in the slip-duplicate treatment had significantly higher phenotypic match scores than organisms evolved in the slip-scramble treatment (two-tailed Mann-Whitney U tests, W = 4305, 4028.5, 3621.5 respectively, Bonferroni-adjusted $p$ values 0.0109, 0.0222, 0.0017, respectively). This result supports that %the capacity to duplicate more information is better, and the fact that gene duplications are capable of
duplicating both the content and structure of existing genetic code is an important factor in their ability to promote evolvability.

\subsection{Discerning the Mechanism of Slip Duplication in Facilitating Adaptive Evolution}

Thus far, we have established that the locality and content of slip-duplicated genetic material both contribute substantially to its effectiveness in facilitating adaptive evolution.
We next sought to understand \textit{how} these mechanisms drive faster adaptive evolution.
In these investigations, we aimed to (1) identify how the effectiveness of gene duplication relates to trait complexity, (2) test for signatures of evolutionary potentiation by gene duplications along lineage histories, and (3) examine how slip duplication influences genetic architecture with respect to brittleness and vestigial genetic material.

\subsubsection{Slip duplication facilitates the evolution of complex traits.}

Within the logic-9 task system in Avida, available tasks vary substantially in their computational difficulty.
In fact, explicit hierarchical composition exists mong task definitions.
``Easier'' tasks like AND and NAND form building blocks for ``more difficult'' tasks like XOR and EQU.
Previous studies have used this property to explore the evolution of complex traits \citep{lenski2003evolution}.
We categorize the complexity of the Logic-9 tasks used in this work by the minimal number of NAND building blocks\footnote{
In computer architecture theory, NAND is commonly established as the fundamental building block operation since all other logical operations can be derived from compositions of NAND gates \citep{TODO}.
} required to perform them:
\begin{enumerate}
\item NOT and NAND: one NAND component,
\item AND and ORNOT: two NAND components,
\item OR and ANDNOT: three NAND components,
\item NOR and XOR: four NAND components, and
\item EQU: five NAND components.
\end{enumerate}
We leveraged this natural hierarchy to test how gene duplication facilitates evolution across a spectrum of task complexities by studying the rate at which tasks with different NAND ``component counts'' evolved.

As explored earlier, one intrinsic consequence of gene duplication is increased genome length.
To account for this effect, we considered an additional control in our experiments: the long-genome baseline treatment, in which organisms employed fixed-length 1,000-site genomes.

\input{fig/adaptive-evolution-rate.tex}

First, we compared the adaptive boost from slip duplication against that from simply having longer genomes.
Figure \ref{fig:adaptive-evolution-rate} shows rates of task acquisition with and without slip duplication, including this long-genome control.

The long-genome control matched or exceeded the performance of slip-duplication in evolving simple traits with 3 or fewer components.
% https://github.com/chaynes2019/AvidaGeneDupe/blob/538ede79c7301f10718ca96c8dd38782b6882632/binder/adaptive-evolution-rate.ipynb
However, slip duplication evolved more complex 4- and 5-component traits in a significantly higher fraction of replicates compared to the long-genome control (Fisher's exact tests; 36/60 vs. 24/60, $p<0.05$ [4 components]; 10/30 vs. 2/30, $p<0.03$ [5 components]; Figure \ref{fig:adaptive-evolution-rate:direct}).
As such, these findings are consistent with the hypothesis of gene duplication as a catalyst for the evolution of complex traits.

\subsubsection{Slip-duplicated genome regions potentiate evolution of novel complex traits.}

% https://github.com/chaynes2019/AvidaGeneDupe/blob/4c7fa27229094adcb5bdb0b1aec541d0014b0fed/binder/hard-task-gain.ipynb
%             H-statistic       p-value
% Components
% 1              1.011820  3.144672e-01
% 2             22.787798  1.809107e-06
% 3             33.846753  5.962854e-09
% 4             15.359894  8.885441e-05
% 5              9.097744  2.559249e-03
%    Components  Prev Slip Insertion Cumulative Count      mean       std
% 0           1                                 False  0.932088  0.455109
% 1           1                                  True  1.002025  1.164197
% 2           2                                 False  0.588566  0.618479
% 3           2                                  True  1.579974  1.191681
% 4           3                                 False  0.491481  0.700256
% 5           3                                  True  1.516115  0.911430
% 6           4                                 False  0.703793  0.980105
% 7           4                                  True  1.229158  0.654042
% 8           5                                 False  0.583674  0.927917
% 9           5                                  True  1.249985  0.531423
%    Components  Prev Slip Insertion Cumulative Count  size
% 0           1                                 False    60
% 1           1                                  True    48
% 2           2                                 False    60
% 3           2                                  True    56
% 4           3                                 False    60
% 5           3                                  True    59
% 6           4                                 False    52
% 7           4                                  True    52
% 8           5                                 False    20
% 9           5                                  True    20

\input{fig/potentiation}

Thus far, we have established that slip duplication facilitates evolution of novel complex traits, we next sought to identify the processes through which slip duplication promotes adaptive evolution.
To this end, we tested whether slip duplication potentiates the evolution of adaptive traits at duplicated genome sites.
For each occurrence of a new trait arising in our experiments, we assessed the fraction of new task coding sites originating in regions of the genome that had previously been slip duplicated.
Figure \ref{fig:potentiation} compares these ratios of involvement in coding for new tasks between previously slip-duplicated and non-slip-duplicated sites.
For the simplest tasks, requiring only one NAND component, we found no significant difference in the likelihood of duplicated sites participating in coding regions for new tasks.
However, we found significant associations for traits with two or more NAND components (Mann-Whitney tests; all $p < 0.01$; $n=48,56,59,52,20$ observations).
Effect sizes of potentiation on likelihood to code for novel traits were $1.5\times$, $1.5\times$, $1.2\times$, and $1.2\times$ respectively for 2, 3, 4, and 5 task components.
Smaller effect sizes at 4- and 5-component tasks may be due to a larger portion of the genome becoming comprised of slip-duplicated sites (Supplementary Figure \ref{fig:potentiation-supp}), thus limiting the ceiling on deviation from expected.

One possible confounding factor in this result is reduced evolutionary rates at genome sites involved in organsims' self-replication loop.
These sites are critical to viability, with lethal outcomes when knocked out.
We found that these critical sites were less likely to be involved in slip duplication and also less likely to be involved in coding for \textit{de novo} traits.
Hence, sensitivity at these sites could introduce a spurious correlation.
However, we found similar potentiation signatures from slip duplication after excluding such fitness-critical slites from analysis (Supplemental Figure \ref{fig:potentiation-supp}).

One alternative to the potentiation hypothesis is that gene duplication directly facilitates adaptation by supplying mutational steps biased to discover beneficial outcomes \citep{TODO}.
Where regularities exist in the genotype-phenotype-fitness map, duplicated variants of existing genetic information may be more likely to encode meaningful traits.
In the fitness landscape analogy, this hypothesis would correspond to steps through fitness space that are larger but less risky.

In line with this possibility, we observed that a substantial fraction of gain-of-function steps on lineages directly coincided with slip duplications (TODO quantify).
However, in these cases, sites that gained a new trait directly due to slip mutation were nonetheless more likely than chance to have been involved in earlier slip duplications (Supplemental Figure \ref{fig:potentiation-supp}).
Thus, the adaptive characteristics of slip duplication seem likely to result from a combination of potentiation and direct facilitation.

\subsubsection{Gene duplication increases accumulation of vestigial coding material rather than decreasing genetic brittleness.}

\input{fig/num-coding-sites}

In a final set of analyses, we investigated broader relationships between gene duplication and genome architecture.

To assess effects on genome robustness, which we considered a definition in terms of brittleness;
that is, the number of genome sites ``critical'' to fitness, where a single-site knockout causes the loss of one or more adaptive phenotypic traits (i.e., tasks).
Single-site knockouts were used to identify critical sites via Avida's genome evaluator.\footnote{%
Although sufficiently representative for our purposes, limitations do exist in detecting Avida genome functionality through single-site knockouts; such an approach can underestimate aspects of genome sequence complexity involving small effects or redundancy \citep{moreno2024cryptic}.
}

We found that, rather than decreasing net brittleness, the slip duplication treatment accrued fitness-critical sites at a generation-on-generation rate comparable to the long-genome baseline treatment (Figure \ref{fig:num-coding-sites}).
However, slip duplications significantly increased the net amount of vestigial and active coding sites accrued in genomes (Figure \ref{fig:num-coding-sites}).
Thus, gene duplication increased supply of coding material in the genome available to neutral processes, even though overall robustness did not increase.
% https://github.com/chaynes2019/AvidaGeneDupe/blob/538ede79c7301f10718ca96c8dd38782b6882632/binder/indel-effect-nulldist.ipynb

In follow-up experiments, we tested the baseline effects of slip insertion mutations on brittleness without selection.
We found that slip insertions that were fitness-neutral tended to reduce net brittleness.
On average, these insertion mutations decreased coding site count by 6.8 sites.
This effect was strongest in genomes with high complexity; for instance, neutral insertion mutations decrease coding site count by 9.2 and 8.3 sites on average in genomes that encode 4- and 5-component complexity tasks, respectively.
Supplementary Figure \ref{fig:nulldist} overviews these results.
% One possible explanation is that selective pressures are driving the two treatments to reach a similar number of critical sites, but slip duplication can duplicate and copy genome content in a manner that increases the copy count of previously coding sites relative to the number of critical sites.



\subsection{Major Results}

We found local slip mutations of intact regions to be the most effective configuration of gene duplication in facilitating evolution on the logic-9 task set within the Avida platform.
In particular, we found that --- compared to control experiments with long genome sizes --- gene duplication uniquely promoted the evolution of complex adaptive traits.
We further found that the raw material created by slip duplication plays a potentiating role in the evolution of complex traits.
Specifically, we identified that slip-duplicated regions are significantly more likely to provide coding sites for new traits when they first appear.
Consistent expectations under neofunctionalization theory, however, we did not observe potentiation effects of slip duplication on the evolution of very simple traits that did not involve building block components.

Finally, we assessed the consequences of slip duplication on genome architecture.
One possible expectation is that gene duplication might accelerate growth in genetic brittleness by providing raw material for contingent complexity to arise, as drift effects wear away the redundancies introduced by duplication.
Contrary to this expectation, we found that the rate of genome complexity growth was similar to that observed in control experiments.
However, we observed a significant increase in the accumulation rate of net active and vestigial coding material in genomes under slip duplication.
To understand this phenomenon, we tested the immediate effects of slip duplication on genome brittleness.
We found that, on average, neutral slip duplications decrease, rather than increase, the number of critical coding sites in a genome.
These results align with our observed increase in vestigial coding material in genomes.
These brittleness-reducing effects appear to be counteracted by other factors, resulting in a similar overall trajectory of genome complexity between slip duplication treatments and controls.

\subsubsection{Outro/pitch.}
Strong theoretical interest exists in characterizing the role of gene duplication in facilitating neutral evolution and enabling adaptive evolution of new complexity through subfunctionalization and neofunctionalization.
Indeed, ample evidence from existing experiments \textit{in silico} and \textit{in vivo} has supported the role of gene duplication in facilitating the evolution of novelty, complexity, and adaptation.
Our study contributes to this conversation by highlighting in greater detail and specificity the consequences of gene duplication in a full-fledged digital model:
\begin{enumerate}
\item While increasing the amount of raw genetic material suffices to facilitate the evolution of simple traits, duplicating existing genome material excels in facilitating the evolution of complex traits;
\item Mechanistically, duplications significantly potentiate genome regions that go on to code for novel complex traits but --- consistent with neofunctionalization theory --- do not potentiate simple traits;
\item From a genetic architecture perspective, we do not find direct evidence of subfunctionalization in increasing genome brittleness under slip duplication; however, slip duplication accelerates the accumulation of net vestigial and active coding material in genomes.
\end{enumerate}

% \subsection{High mutation rate inhibits the evolution of regulation in changing environments}
% In addition to our results on why gene duplications promote evolvability, our experiments
% %demonstrate the potential for
% indicate that
% high mutation rates inhibit the evolution of regulation in changing environments. In both the simple and complex changing environments, organisms evolved in the high mutation rate treatment had significantly lower phenotypic match scores than organisms evolved in the baseline treatment, which can be seen in Figure \ref{fig:results_panels} (two-tailed Mann-Whitney U tests, W = 9958 and 9944, respectively, Bonferroni-adjusted $p$ values both $<< 0.0001$).
% A similar result was found in \citep{Lalejini:2016plasticity}. Although higher mutation rates increase genetic variation, most mutations have deleterious effects \citep{Schlichting:2002jl}. Thus, higher mutation rates may increase the difficulty of maintaining the genetic machinery necessary for complex regulation.

% If elevated mutation rates %can work to
% inhibit the evolution of regulation and all experimental treatments with slip mutation operators have an effectively higher mutation rate than the baseline treatment, why do we see this effect only in the high mutation rate treatment? The rates of insertions and deletions per instruction copied in the high mutation rate treatment were selected to, on average, result in approximately the same number of mutations per divide as in our treatments with slip mutations. Yet, only organisms evolved in the high mutation rate treatment had significantly lower phenotypic match scores than organisms evolved in the baseline treatment in the changing environments; all other experimental treatments were either significantly better or not significantly different than the baseline treatment.

% @amlalejini old text on mutation load explanation:
%Why is it that \textit{only} the high mutation rate treatment inhibited evolvability in the changing environments? One possibility is because slip mutations result in large mutational events that affect few offspring, whereas mutations in the high mutation rate treatment result in an increased number of smaller mutational events that are spread across many offspring. Thus, many offspring in the high mutation rate treatment are subject to the increased rate of deleterious mutations that results from an increase in the rates of copy insertions and copy deletions. This is in contrast to treatments with slip mutations where relatively fewer offspring are subjected to the increased rate of deleterious mutations because deleterious mutations are concentrated in a fewer offspring relative to the high mutation rate treatment.
%Further analysis is needed to confirm that this is, indeed, the case.
% @amlalejini new text on mutation load explanation:
%  - should I cite something for 'mutation load' (i.e. this concept exists, read about it here)?
% Why is it that \textit{only} the high mutation rate treatment inhibited evolvability in the changing environments?
% One possibility stems from the higher mutation load imposed on populations by the high mutation rate treatment.
% Slip mutations result in large mutational events that affect few offspring, whereas the elevated rate of copy mutations in the high mutation rate treatment results in an increased number of smaller mutational events spread across many offspring.
% Thus, many offspring in the high mutation rate treatment are subject to the increased rate of deleterious mutations that results from the higher rates of copy insertions and copy deletions. This is in contrast to treatments with slip mutations where relatively fewer offspring are subjected to large mutational events, which results in the concentration of deleterious mutations in fewer offspring relative to the high mutation rate treatment.
% Further analysis is needed to confirm that this is, indeed, the case.

% OLD figures
% \input{fig/FullReplicationTaskCountTimeCourse.tex}
% \input{fig/FullReplicationFinalDominantTaskCountViolin.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantInfo.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantInfoConcentration.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantGenomeLengthsViolin.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantCodingSitesPerTask.tex}
% \input{fig/ModifiedReplicationFinalDominantTaskCountViolin.tex}
% \input{fig/LalejiniEtAlReplicationRealFinalDominantCodingSitesPerTask.tex}
% \input{fig/LalejiniEtAlReplicationFinalDominantOverlap.tex}
